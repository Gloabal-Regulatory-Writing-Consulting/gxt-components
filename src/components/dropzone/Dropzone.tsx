import { useDropzone } from "react-dropzone";
import { Button } from "../button";
import {
  DropzoneLabel,
  DropzoneMessage,
  DropzoneWrapper,
} from "./Dropzone.styles";

export type DropzoneProps = {
  onDrop?: (acceptedFiles: File[]) => void;
  maxFiles?: number;
  onError?: (message: string) => void;
  label?: string;
};

const Dropzone = ({ onDrop = () => {}, maxFiles = 10, onError = () => {}, label = '' }: DropzoneProps) => {
  const handleDrop = (acceptedFiles: File[]) => {
    if (acceptedFiles.length > maxFiles) {
      onError(`You can only upload ${maxFiles} files`);
      return;
    }
    onDrop(acceptedFiles);
  };

  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop: handleDrop,
  });

  return (
      <DropzoneWrapper {...getRootProps()} data-testid="dropzone">
        <input {...getInputProps()} />
        <DropzoneLabel>
          {isDragActive ? (
            <DropzoneMessage>Drop here ...</DropzoneMessage>
          ) : (
            <>
              <DropzoneMessage>{ label || "Drag and Drop Files"}</DropzoneMessage>
              <Button variant="secondary">or Browse</Button>
            </>
          )}
        </DropzoneLabel>
      </DropzoneWrapper>
  );
};
export default Dropzone;
