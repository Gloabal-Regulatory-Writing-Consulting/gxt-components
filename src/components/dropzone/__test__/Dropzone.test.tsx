import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import Dropzone from "../Dropzone";
import React from "react";

describe("Dropzone component", () => {
  it("should render without errors", () => {
    render(<Dropzone />);
    expect(screen.getByTestId("dropzone")).toBeInTheDocument();
    expect(screen.getByTestId("dropzone")).toMatchSnapshot();
  });
});

export default Dropzone;
