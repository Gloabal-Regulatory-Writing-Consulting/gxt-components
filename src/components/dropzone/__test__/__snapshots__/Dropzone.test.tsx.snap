// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropzone component > should render without errors 1`] = `
<div
  class="sc-iGgVNO cBMcyP"
  data-testid="dropzone"
  role="presentation"
  tabindex="0"
>
  <input
    multiple=""
    style="display: none;"
    tabindex="-1"
    type="file"
  />
  <label
    class="sc-gsFSjX bqmBIO"
  >
    <p
      class="sc-kAycRU loTMNo"
    >
      Drag and Drop Files
    </p>
    <button
      class="sc-aYaIB sc-eqUzNf jTlVdY ajZRB"
    >
      or Browse
    </button>
  </label>
</div>
`;
